azkaban-common/src/main/java/azkaban/user/Permission.java

```diff
@@ -191,6 +191,7 @@ public int toFlags() {
     SCHEDULE(0x0000008),
     METRICS(0x0000010),
     CREATEPROJECTS(0x40000000), // Only used for roles
+    UPLOADPROJECTS(0x0008000),
```

Comment:
If I understand it correctly, this permission will only take effect when lockdown.upload.projects is set to true.

I found it not intuitive since other permissions are not affected by global settings. 

Is this addition necessary? If lockdown.upload.projects is set to true, what if we only allow admins to upload. Is the intention to limit the uploader's permission?

Can you think of alternative design that wouldn't have this dependency on a global setting?